go/enclave/events/subscription_manager.go

```diff
@@ -71,14 +76,18 @@ func (s *SubscriptionManager) AddSubscription(id gethrpc.ID, encryptedSubscripti
 	// We set this to the current rollup height, so that historical logs aren't returned.
 	subscription.Filter.FromBlock = big.NewInt(0).Add(s.storage.FetchHeadRollup().Number(), big.NewInt(1))
 
+	s.subscriptionMutex.Lock()
```

Comment:
I think I see this most commonly with the Unlock being deferred immediately on the line below the Lock, like:
```
mut.Lock()
defer mut.Unlock()
```
even for simple examples like this to avoid risk of someone creating a bug later.

Reply:
agree. I think we should do this everywhere, unless there is a very good reason not to